tests/functional/test_wps_provider.py

```diff
@@ -0,0 +1,248 @@
+import contextlib
+from typing import TYPE_CHECKING
+
+import mock
+import pyramid.testing
+import pytest
+
+from tests import resources
+from tests.functional.utils import WpsConfigBase
+from tests.utils import (
+    mocked_execute_process,
+    mocked_remote_server_requests_wps1,
+    mocked_sub_requests,
+    setup_mongodb_jobstore,
+    setup_mongodb_processstore,
+    setup_mongodb_servicestore
+)
+from weaver.config import WEAVER_CONFIGURATION_HYBRID
+from weaver.execute import (
+    EXECUTE_CONTROL_OPTION_ASYNC,
+    EXECUTE_MODE_ASYNC,
+    EXECUTE_RESPONSE_DOCUMENT,
+    EXECUTE_TRANSMISSION_MODE_REFERENCE
+)
+from weaver.formats import CONTENT_TYPE_APP_NETCDF, CONTENT_TYPE_TEXT_PLAIN, CONTENT_TYPE_TEXT_XML
+from weaver.processes.types import PROCESS_WPS_REMOTE
+from weaver.processes.wps1_process import Wps1Process
+
+if TYPE_CHECKING:
+    from responses import RequestsMock
+
+    from tests.utils import MockPatch
+
+
+# pylint: disable=C0103,invalid-name
+@pytest.mark.functional
+class WpsProviderTest(WpsConfigBase):
+    settings = {
+        # NOTE: important otherwise cannot execute "remote" provider (default local only)
+        "weaver.configuration": WEAVER_CONFIGURATION_HYBRID,
+        "weaver.wps_output_dir": "/tmp",
+        "weaver.wps_output_url": resources.TEST_REMOTE_SERVER_URL + "/wps-outputs"
+    }
+
+    @classmethod
+    def tearDownClass(cls):
+        pyramid.testing.tearDown()
+
+    def setUp(self):
+        # rebuild clean db on each test
+        self.service_store = setup_mongodb_servicestore(self.config)
+        self.process_store = setup_mongodb_processstore(self.config)
+        self.job_store = setup_mongodb_jobstore(self.config)
+        self.app_url = self.settings["weaver.url"]
+
+    @mocked_remote_server_requests_wps1([
+        resources.TEST_REMOTE_SERVER_URL,
+        resources.TEST_INVALID_ESCAPE_CHARS_GETCAP_WPS1_XML,
+        [],
+    ])
+    def test_register_finch_with_invalid_escape_chars(self):
+        """
+        Remote provider that has invalid characters that should be escaped (<, <=) but should succeeds registration.
+
+        This test ensure that temporary patch of XML parser with *permissive* invalid characters that can be recovered
```

Comment:
What about making this a config option rather than a temporary patch?
It might not fix the root cause of the problem, but it's an alternative.

Reply:
It is complicated to make this dynamic. 
To make this work, I have to substitute (monkey-patch) the XML parser within OWSLib and PyWPS. 
I prefer to apply the same parser each time to avoid making this more complicated than it should.

Reply:
I see. In our case that's not an issue since we use PyWPS which we know the fix is on the way.
When this temporary logic will get removed, maybe it could be a good thing to make it more specific to the user that the root cause is an invalid XML, rather than `Provider properties could not be parsed correctly`. It would be clear that the issue came from the XML itself, and not another provider property.

Reply:
I will add some handling to attempt better reporting that error is related to XML parsing vs something else.

Reply:
@matprov 
See added tests in ab6495a for examples of error messages that will now be returned for different cases.

Reply:
+1, I like the "recover partially bad XML" feature that you've added